presto-product-tests/src/main/java/io/prestosql/tests/kafka/TestKafkaAvroWritesSmokeTest.java

```diff
@@ -0,0 +1,153 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.kafka;
+
+import com.google.common.collect.ImmutableList;
+import io.prestosql.tempto.ProductTest;
+import io.prestosql.tempto.Requirement;
+import io.prestosql.tempto.RequirementsProvider;
+import io.prestosql.tempto.Requires;
+import io.prestosql.tempto.configuration.Configuration;
+import io.prestosql.tempto.fulfillment.table.kafka.KafkaTableDefinition;
+import io.prestosql.tempto.fulfillment.table.kafka.ListKafkaDataSource;
+import org.testng.annotations.Test;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static io.prestosql.tempto.assertions.QueryAssert.Row.row;
+import static io.prestosql.tempto.assertions.QueryAssert.assertThat;
+import static io.prestosql.tempto.fulfillment.table.TableRequirements.immutableTable;
+import static io.prestosql.tempto.query.QueryExecutor.query;
+import static io.prestosql.tests.TestGroups.KAFKA;
+import static io.prestosql.tests.TestGroups.PROFILE_SPECIFIC_TESTS;
+import static java.lang.String.format;
+import static org.testng.Assert.assertTrue;
+
+public class TestKafkaAvroWritesSmokeTest
+        extends ProductTest
+{
+    private static final String KAFKA_CATALOG = "kafka";
+
+    private static final String ALL_DATATYPES_AVRO_TABLE_NAME = "product_tests.write_all_datatypes_avro";
+    private static final String ALL_DATATYPES_AVRO_TOPIC_NAME = "write_all_datatypes_avro";
+
+    private static final String ALL_NULL_AVRO_TABLE_NAME = "product_tests.write_all_null_avro";
+    private static final String ALL_NULL_AVRO_TOPIC_NAME = "write_all_null_avro";
+
+    private static final String STRUCTURAL_AVRO_TABLE_NAME = "product_tests.write_structural_datatype_avro";
+    private static final String STRUCTURAL_AVRO_TOPIC_NAME = "write_structural_datatype_avro";
+
+    // kafka-connectors requires tables to be predefined in presto configuration
+    // the requirements here will be used to verify that table actually exists and to
+    // create topics and propagate them with data
+    private static class AllDataTypesAvroTable
+            implements RequirementsProvider
+    {
+        @Override
+        public Requirement getRequirements(Configuration configuration)
+        {
+            return immutableTable(new KafkaTableDefinition(
+                    ALL_DATATYPES_AVRO_TABLE_NAME,
+                    ALL_DATATYPES_AVRO_TOPIC_NAME,
+                    new ListKafkaDataSource(ImmutableList.of()),
+                    1,
+                    1));
+        }
+    }
+
+    @Test(groups = {KAFKA, PROFILE_SPECIFIC_TESTS})
+    @Requires(AllDataTypesAvroTable.class)
+    public void testInsertPrimitiveDataType()
+    {
+        assertThat(query(format(
+                "insert into %s.%s values " +
```

Comment:
can we extend the `ALL_DATATYPES_AVRO_TABLE_NAME` to cover also:
* BIGINT column with Avro type set to INT
* REAL column with Avro type set to FLOAT
* DOUBLE column with Avro type set to DOUBLE?

Can be a followup

Reply:
Logged as https://github.com/prestosql/presto/issues/5952 for follow-up.